tests/subsys/dfu/dfu_target/smp/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(dfu_target_smp_test)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources})
+
+target_sources(app
+  PRIVATE
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/dfu/dfu_target/src/dfu_target_smp.c
+  )
+
+set(includes
+"src/"
+)
+
+target_include_directories(app
+  PRIVATE
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/dfu/include
+  ${includes}
+  )
+
+target_compile_options(app
+  PRIVATE
+  -DCONFIG_IMG_BLOCK_BUF_SIZE=4096
```

Comment:
is there a reason why these are provided through Cmake and not through the YAML?

Reply:
These need to define here that build is not failing. Enabling CONFIG_TARGET_SMP=y was causing a the other problem. We have done similar for other Unit test also.

Reply:
then if it fails for a specific platform you'll have to just filter the allow list.